credentials/tls/certprovider/pemfile/watcher_test.go

```diff
@@ -302,82 +349,102 @@ func (s) TestProvider_UpdateSuccess(t *testing.T) {
 		}
-	}
-	km1, err := prov.KeyMaterial(ctx)
-	if err != nil {
-		t.Fatalf("provider.KeyMaterial() failed: %v", err)
-	}
+		t.Run(testName, func(t *testing.T) {
+			// Override the newDistributor to one which pushes on a channel that we
+			// can block on.
+			origDistributorFunc := newDistributor
+			distCh := testutils.NewChannel()
+			d := newWrappedDistributor(distCh)
+			newDistributor = func() distributor { return d }
+			defer func() { newDistributor = origDistributorFunc }()
+
+			// Create two tempDirs with different files.
+			dir1 := createTmpDirWithFiles(t, "update_with_symlink1_*", "x509/client1_cert.pem", "x509/client1_key.pem", "x509/client_ca_cert.pem", "spiffe/spiffebundle.json")
+			dir2 := createTmpDirWithFiles(t, "update_with_symlink2_*", "x509/server1_cert.pem", "x509/server1_key.pem", "x509/server_ca_cert.pem", "spiffe/spiffebundle2.json")
+
+			// Create a symlink under a new tempdir, and make it point to dir1.
+			tmpdir, err := os.MkdirTemp("", "test_symlink_*")
+			if err != nil {
+				t.Fatalf("os.MkdirTemp() failed: %v", err)
+			}
+			symLinkName := path.Join(tmpdir, "test_symlink")
+			if err := os.Symlink(dir1, symLinkName); err != nil {
+				t.Fatalf("failed to create symlink to %q: %v", dir1, err)
+			}
 
-	// Update the symlink to point to dir2.
-	symLinkTmpName := path.Join(tmpdir, "test_symlink.tmp")
-	if err := os.Symlink(dir2, symLinkTmpName); err != nil {
-		t.Fatalf("failed to create symlink to %q: %v", dir2, err)
-	}
-	if err := os.Rename(symLinkTmpName, symLinkName); err != nil {
-		t.Fatalf("failed to update symlink: %v", err)
-	}
+			// Create a provider which watches the files pointed to by the symlink.
+			var opts Options
+			if useSPIFFEBundle {
+				opts = Options{
+					CertFile:            path.Join(symLinkName, certFile),
+					KeyFile:             path.Join(symLinkName, keyFile),
+					RootFile:            path.Join(symLinkName, rootFile),
+					SPIFFEBundleMapFile: path.Join(symLinkName, spiffeBundleFile),
+					RefreshDuration:     defaultTestRefreshDuration,
+				}
+			} else {
+				opts = Options{
+					CertFile:        path.Join(symLinkName, certFile),
+					KeyFile:         path.Join(symLinkName, keyFile),
+					RootFile:        path.Join(symLinkName, rootFile),
+					RefreshDuration: defaultTestRefreshDuration,
+				}
+			}
+			prov, err := NewProvider(opts)
+			if err != nil {
+				t.Fatalf("NewProvider(%+v) failed: %v", opts, err)
+			}
+			defer prov.Close()
+
+			// Make sure the provider picks up the files and pushes the key material on
+			// to the distributors.
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+			for i := 0; i < 2; i++ {
+				// Since we have root and identity certs, we need to make sure the
+				// update is pushed on both of them.
+				if _, err := distCh.Receive(ctx); err != nil {
+					t.Fatalf("timeout waiting for provider to read files and push key material to distributor: %v", err)
```

Comment:
Here and in other places in this PR, test errors need not follow the error string capitalization guideline. Please see: go/go-style/decisions#error-strings

Reply:
I apologize, a lot of this is just code that is in the diff because it was tabbed over, I didn't closely read all of the existing errors. I'll do my best to clean up